Payjoin-spec.md

```diff
@@ -0,0 +1,182 @@
+
+## Implementation notes
+
+### Receiver side
+
+The BTCPayServer implementation is making the following checks on the original PSBT:
+
+* The original PSBT must be finalized (Fails with error `psbt-not-finalized`)
+* All the inputs must be P2WPKH (Fails with error `unsupported-inputs`) [*](#unsupported-inputs)
+* The PSBT is sane (Fails with error `insane-psbt`)
+* All inputs have the `witnessUTXO` (Fails with error `need-utxo-information`)
+* Check that there is no global xpubs, HD Key or public key information in the PSBT (Fails with error `leaking-data`)
+* Check that the extracted transaction can be accepted by mempool (Fails with error `invalid-transaction`)
+
+Then the original PSBT is deemed valid, however, other checks are done which depends on the state of the receiver's state.
+
+* Check that it pays an actual invoice (Fails with error `invoice-not-found`)
+* Check that the invoice does not have any outstanding payment (Fails with error `already-paid`)
+* Check that the whole amount of the invoice is paid (Fails with error `invoice-not-fully-paid`)
+* Check that the inputs used in the original PSBT have never been used for another payjoin (Fails with `inputs-already-used`)
+* Check that the receiver has available UTXO to contribute to the payjoin (Fails with error `out-of-utxos`)
+* Check that the receiver can actually sign (Fails with error `unavailable`)
+
+Finally, we need to increase the fees to pay for the increased size of payjoin transaction.
+From this point, the receiver will automatically try to broadcast the original transaction after 1 minute, whatever happens.
+
+* If the sender overpaid the invoice, we substract fees on the overpaid part of the output paying the invoice. [**](#poisoning)
+* We substract the rest to other outputs, making sure their value does not go below dust.
+* If there is not enough money to pay for the increased fee, check that the fee rate of the payjoin transaction is above the minimum relay fee (today 1 satoshi per byte). If not, fails with error `not-enough-money`.
+
+
+\* <a name="unsupported-inputs"></a>: We currently only support receivers and senders using P2WPKH, while we plan to relax this in a later release, it is important for the receiver to use the same input's type as the sender, otherwise the payjoin transaction would provide no privacy benefit. Since most wallet do not support P2WSH or wrapped P2SH for the sender, we decided to not offer this feature for now.
+
+\*\* <a name="poisoning"></a>: If the sender overpays the invoice, we can substract fees from his over paid amount. This is especially useful when somebody wants to make a donation from a small unspent output in his own wallet.
+
+### Sender side
+
+Before sending submitting the original PSBT to the receiver, the sender make sure that the original PSBT is broadcasted whatever happen by automatically broadcasting it after 1 minute.
```

Comment:
`Before sending submitting the original PSBT to the receiver, the sender make sure that the original PSBT is broadcasted whatever happen by automatically broadcasting it after 1 minute` => 

`Before sending the original PSBT to the receiver, the sender should make sure that the original PSBT is at least attempted to be broadcasted automatically after 1 minute of submission`